Please enter the commit message for your changes. Lines starting

SvetaHBar opened the pull request.

SvetaHBar committed:
Please enter the commit message for your changes. Lines starting
Introduce new cactus module

EdDev commented:
Please fix the commit message and then the poll request name.

EdDev commented on src/cactus.py:
```diff
@@ -0,0 +1,52 @@
+    def __init__(self, kind, watering):
+        self._kind = kind
+        self._watering = watering
+        self._is_succulent = True
```
What is this suppose to represent? A kind/type?

EdDev commented on src/cactus.py:
```diff
@@ -0,0 +1,52 @@
+
+    def __init__(self, kind, watering):
+        self._kind = kind
+        self._watering = watering
```
Why "watering" is an inputted value? It does not fit as a default value?

EdDev commented on src/cactus.py:
```diff
@@ -0,0 +1,52 @@
+        self._watering = watering
+        self._is_succulent = True
+        self._is_plant_dry = True
+        self._is_plant_humid = True
```
It is dry and humid at the same time?

EdDev commented on src/cactus.py:
```diff
@@ -0,0 +1,52 @@
+    def is_succulent(self):
+        return self._is_succulent
+
+    def need_to_water(self):
```
One function should do the operation of water (to water).
A second function should ask if it is time to water.

When you perform the water operation, you stamp the current time (time.time() is good enough) on the cactus object. Then, when you ask if it is time to water it, it should get the current time, subtract the one saved on the object when last water operation occurred and calculate the diff.
Note that time() returns seconds, do the diff will be in seconds.

SvetaHBar committed:
adding two changes with None error
adding new tests if_need_to _water
change def need_to_water , added new tests
added new tests large_potter and need_shadow or need_sun
edit the cactus crs and tests_cactus with edy's comments

SvetaHBar commented:
Please review

EdDev commented on src/cactus.py:
```diff
@@ -0,0 +1,82 @@
+        else:
+            return True
+    # more optimated way to write "if" :
+            #return delta_time > 7777
```
You should use this version, not the noisy one.

EdDev commented on src/cactus.py:
```diff
@@ -0,0 +1,82 @@
+    def need_to_water(self):
+
+        current_time = time.time()
+        print current_time
```
No prints in production code please.

EdDev commented on src/cactus.py:
```diff
@@ -0,0 +1,82 @@
+
+    def __init__(self, kind):
+        self._kind = kind
+        #self._watering = watering
```
Remove the leftovers please.

EdDev commented on src/cactus.py:
```diff
@@ -0,0 +1,82 @@
+        self._is_plant_humid = False
+        self.watering_time = 0
+        self.potter = 0
+        self.is_need_sun = True
```
Drop the 'is', it does not fit.

EdDev commented on src/cactus.py:
```diff
@@ -0,0 +1,82 @@
+        self._is_succulent = True
+        self._is_plant_dry = True
+        self._is_plant_humid = False
+        self.watering_time = 0
```
Needs to be private, including others below I guess.

EdDev commented on src/cactus.py:
```diff
@@ -0,0 +1,82 @@
+        #return self._is_plant_dry
+
+    def is_need_large_potter(self):
+        current_potter = time.time()
```
you mean current_time, right?

EdDev commented on src/cactus.py:
```diff
@@ -0,0 +1,82 @@
+        self.is_need_sun = False
+
+
+def foo():
```
Does not belong here, probably should be dropped.
